gui/mozregui/pref_editor.py

```diff
@@ -30,13 +30,13 @@ def data(self, index, role=Qt.DisplayRole):
             if index.column() == 0:
                 return name
             else:
-                return repr(value)
+                return QString(value).__str__()
 
     def flags(self, index):
         return Qt.ItemIsSelectable | Qt.ItemIsEditable | Qt.ItemIsEnabled
 
     def setData(self, index, new_value, role=Qt.EditRole):
-        new_value = str(new_value.toString())
+        new_value = new_value.toPyObject().__str__()
```

Comment:
I believe this is the same, that is new_value will still be a 'str' object, not 'unicode', no ? I think just replacing str(new_value.toString()) to **unicode**(new_value.toString()) will work well. You can try it by editing a cell with a non unicode character (like "é").